### ISSUE TITLE
Improve memory usage of uv_poll_t

### ISSUE THREAD

**Issue by alexhultman** [libuv/libuv#1095]
Hey I just made a proof of concept implementation of the uv_poll_t component of libuv and I was able to reduce the memory usage from 160 bytes per poll to 32 bytes. Performance seems to also be better.

This resulted in 1 million websockets per 202 mb of user space memory as compared to 355 mb with the current official uv_poll_t implementation. It seems to be a fair amount of room for improvement of the official uv_poll_t.

**Comment by txdv** [comment]
Where is the code?

**Comment by alexhultman** [comment]
I just made a very basic prototype to test with. It basically just wraps epoll (I only made a Linux prototype). I get that 32 bytes might be a little extreme but 160 should definitely be open for improvements.

**Comment by saghul** [comment]
We wrap the polling mechanism with uv__io_t, I'm not sure we want to get
away from that and repeat the code... But even an early patch may show
where you're going and if it would be acceptable, so an early PR would be
welcome.

On Oct 15, 2016 19:03, "Alex Hultman" <notifications@github.com> wrote:

I just made a very basic prototype to test with. It basically just wraps
epoll (I only made a Linux prototype). I get that 32 bytes might be a
little extreme but 160 should definitely be open for improvements.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/libuv/libuv/issues/1095#issuecomment-253996789>, or mute
the thread
